var/spack/repos/builtin/packages/pism/package.py

```diff
@@ -0,0 +1,175 @@
+
+    version('0.7.x', git='https://github.com/pism/pism.git',
+        branch='stable0.7')
+
+    version('icebin', git='https://github.com/pism/pism.git',
+        branch='efischer/dev')
+
+    version('develop', git='https://github.com/pism/pism.git',
+        branch='dev')
+
+    variant('extra', default=False,
+            description='Build extra executables (testing/verification)')
+    variant('shared', default=True,
+            description='Build shared Pism libraries')
+    variant('python', default=False,
+            description='Build python bindings')
+    variant('icebin', default=False,
+            description='Build classes needed by IceBin')
+    variant('proj', default=True,
+            description='Use Proj.4 to compute cell areas, '
+            'longitudes, and latitudes.')
+    variant('parallel-netcdf4', default=False,
+            description='Enables parallel NetCDF-4 I/O.')
+    variant('parallel-netcdf3', default=False,
+            description='Enables parallel NetCDF-3 I/O using PnetCDF.')
+    variant('parallel-hdf5', default=False,
+            description='Enables parallel HDF5 I/O.')
+    # variant('tao', default=False,
+    #         description='Use TAO in inverse solvers.')
+
+    description = 'Build PISM documentation (requires LaTeX and Doxygen)'
+    variant('doc', default=False, description=description)
+
+    variant('examples', default=False,
+            description='Install examples directory')
+
+    description = 'Report errors through Everytrace (requires Everytrace)'
+    variant('everytrace', default=False, description=description)
+
+    # CMake build options not transferred to Spack variants
+    # (except from CMakeLists.txt)
+    #
+    # option (Pism_TEST_USING_VALGRIND "Add extra regression tests
+    #         using valgrind" OFF)
+    # mark_as_advanced (Pism_TEST_USING_VALGRIND)
+    #
+    # option (Pism_ADD_FPIC "Add -fPIC to C++ compiler flags
+    #         (CMAKE_CXX_FLAGS). Try turning it off if it does not work." ON)
+    # option (Pism_LINK_STATICALLY
+    #         "Set CMake flags to try to ensure that everything is
+    #         linked statically")
+    # option (Pism_LOOK_FOR_LIBRARIES
+    #         "Specifies whether PISM should look for libraries. (Disable
+    #         this on Crays.)" ON)
+    # option (Pism_USE_TR1
+    #        "Use the std::tr1 namespace to access shared pointer
+    #        definitions. Disable to get shared pointers from the std
+    #        namespace (might be needed with some compilers)." ON)
+    # option (Pism_USE_TAO "Use TAO in inverse solvers." OFF)
+
+    depends_on('fftw')
+    depends_on('gsl')
+    depends_on('mpi')
+    depends_on('netcdf')    # Only the C interface is used, no netcdf-cxx4
+    depends_on('petsc')
+    depends_on('udunits2')
+    depends_on('proj')
+    depends_on('everytrace', when='+everytrace')
+
+    extends('python', when='+python')
+    depends_on('python@2.7', when='+python')
```

Comment:
There is no Python 2.7. If it only works with Python 2.7.X, you'll need to use `python@2.7:2.8`. Otherwise, Spack will literally try to fetch and install `python@2.7`.